Support to match private methods/properties by keywords

KarolSiedlik opened the pull request:
Methods and properties are now checked against their accessibility, allowing to sort them by this value. #56

bryanrsmith commented on README.md:
```diff
@@ -30,10 +30,14 @@ Add `sort-class-members` to the plugins section of your `.eslintrc` configuratio
 				"order": [
 					"[static-properties]",
 					"[static-methods]",
+					"[publicProperties]",
```
Can we make the casing on these match the existing convention?

KarolSiedlik replied:
Fixed, thanks!

bryanrsmith reviewed (changes requested):
Thanks for the PR! It looks good, but could you please add some test coverage on the new functionality?

KarolSiedlik commented:
Added unit tests to cover this feature. Replaced default parser in RuleTester with `@typescript-eslint/parser` to allow testing public/private properties. It should not affect the current test cases.

KarolSiedlik commented:
The schema was updated with a recent private field.

bryanrsmith commented on test/rules/sort-class-members.spec.js:
```diff
@@ -707,5 +729,33 @@ ruleTester.run('sort-class-members', rule, {
+		// Public and private methods
+		{ 
+			code: 'class A { private a(){} b(){} c(){} }',
+			output: 'class A { b(){} private a(){}  c(){} }',
```
Interesting. This looks like it's adding support for TypeScript's `private` keyword instead of [JavaScript's private class members](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields). I'm unsure if these things should be conflated. Does TS produce an AST that's compatible with what you get with native private syntax? i.e., if you add tests that use JS's `#`  syntax, do they pass?

KarolSiedlik replied:
They don't. I don't have any expertise with parsers etc., but without `@typescript-eslint/parser` tests were throwing an error "unexpected token: c" when `c` function was marked as private. Now when using `#` rule tester sees no error in the given code. I assume this public/private works only for typescript keywords, so I might change naming to `ts-private-methods`, `ts-public-properties`.......

KarolSiedlik replied:
any updates on that?

bryanrsmith replied:
I'm a bit wary about introducing confusion with ECMA private class fields feature. I'd prefer not to merge support for TS private modifiers without support for ECMA private class fields. Also, if we add support for the `private` modifier, we should also add support for the other accessibility modifiers. I agree that TS-specific features should be labeled as such. I think my preference would be:

* Replace the `private` matcher with a `typescript-accessibility` matcher, accepting a string. e.g., instead of `private: true`, it would be `typescript-accessibility: 'private'` (also accepting `public` and `protected`).
* Use the `private` matcher to match ECMA private class fields.

Thoughts?

KarolSiedlik replied:
Looks good, I will try to implement it in my free time.

bryanrsmith commented:
Closing due to inactivity. Support for matching ecma private fields was added #74.
